app/src/main/java/com/example/movieapp/ui/viewmodel/MainViewModel.kt

```diff
@@ -64,35 +69,44 @@ class MainViewModel : ViewModel() {
         get() = _currentList
 
     init {
-        _currentList.addSource(likedList) { listMovies ->
 
-            val bufM = movieList.value
-            if (bufM is SuccessMoviePrepared<List<Pair<Movie, Boolean>>>) {
-                _currentList.value = SuccessMoviePrepared(bufM.content.map {
-                    Pair(it.first, listMovies.contains(it.first))
-                })
-            }
-        }
+        Log.d(TAG, "initialisation of your mainviewmodel: ")
 
         _currentList.addSource(movieList) { listMoviePrepared ->
+            Log.d(TAG, "movie list is changed: ")
             _currentList.value =
                 if (listMoviePrepared is SuccessMoviePrepared<List<Pair<Movie, Boolean>>>) {
+
                     SuccessMoviePrepared(listMoviePrepared.content.map {
-                        Pair(it.first, likedList.value?.contains(it.first) ?: false)
+
+                        Pair(it.first, likedList.value!!.contains(it.first))
                     })
                 } else {
+                    Log.i(TAG, "list movie is not a succesMoviePrepared: ")
+
                     listMoviePrepared
                 }
         }
     }
 
-
-
-
     fun getList(typeDisplay: TypeDisplay) {
         this._typeDisplay.value = typeDisplay
     }
 
+    fun refresh() {
```

Comment:
I don't know what was your issue, but it seems to be an ugly hack 😅. your viewModel is doing "getAllMovies()" twice